packages/firestore/src/remote/persistent_stream.ts

```diff
@@ -361,6 +381,22 @@ export abstract class PersistentStream<
    */
   protected tearDown(): void {}
 
+  /**
+   * Cancels idel-close-timer and reset backoff as necessary. Mainly those
+   * stream-specific close operations.
+   * @param finalState The intended state of the stream after closing.
+   */
+  private closeWithFinalState(finalState: PersistentStreamState): void {
```

Comment:
This method doesn't really do what it claims to do: the caller still has a bunch of logic in it related to adjusting/reseting the backoff.

I also don't understand this comment at all: "Mainly those stream-specific close operations".

Also note that the iOS method named `closeWithFinalState` already has an equivalent here: it's `close` so this is misleadingly named.

Reply:
removed